pkgs/by-name/di/dita-ot/package.nix

```
@@ -0,0 +1,62 @@
+{ stdenv
+, fetchzip
+, bash
+, jdk17
+, nix
+, coreutils
+, lib
+, makeWrapper
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "dita-ot";
+  version = "4.2.1";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+  buildInputs = [
+    bash
+    jdk17
+  ];
+
+  src = fetchzip {
+    url = "https://github.com/dita-ot/dita-ot/releases/download/${finalAttrs.version}/dita-ot-${finalAttrs.version}.zip";
+    hash = "sha256-PtscKsQwW1uBu0OfVHi7/d9JujGx12nZT+W3uMPQQWc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/$name/
+    cp -r $src/* $out/$name/
+    chmod -R 777 $out/$name/
+
+    wrapProgram "$out/$name/bin/dita" \
+      --prefix PATH : "${lib.makeBinPath [ jdk17 ]}" \
+      --set-default JDK_HOME "$jdk17" \
+      --set-default JAVA_HOME "$jdk17" \
+      --set-default ANDROID_HOME "$jdk17"
+
+    ln -sf "$out/$name/bin/dita" "$out/bin/dita"
```

You could also take a little shortcut (i.e. you won’t need the symlink) by using `makeWrapper`:

```suggestion
    makeWrapper $out/$name/bin/dita $out/bin/dita \
      --prefix PATH : "${lib.makeBinPath [ jdk17 ]}"
```

Also, since the `dita` startup script uses `java` if one is in `PATH`, the environment variables are not actually needed.

---

Implemented 👍🏼